### PR TITLE
Rewrite all code in disspy

### LINKED ISSUE #91: Edit `NoReturn` to `None`
Edit `NoReturn` to `None` in all code

### LINKED ISSUE #93: Rewrite all code
Checklist:
- [ ] abstract.py
- [ ] activity.py
- [ ] application_commands.py
- [ ] channel.py
- [ ] client.py
- [ ] core.py
- [ ] embed.py
- [ ] errors.py
- [ ] guild.py
- [ ] jsongenerators.py
- [ ] reaction.py
- [ ] thread.py
- [ ] typ.py
- [ ] ui.py
- [ ] user.py
- [ ] Tests
- [ ] Optimizations
- [ ] PyLint fixes

### LINKED ISSUE #96: Update application commands
**Description of your idea**

Type here

**Ideal solution**
```python
@bot.command() # /test
@app_commands.describe("This is example command")
@app_commands.options.describe(message=Option(type=3, required=True))
async def test(ctx: Context):
    await ctx.send(ctx.args.get("message"))
```
**Not ideal solution**
```python
@bot.command(description="This is example command", message=Option(type=3, required=True))
async def test(ctx: Context):
    await ctx.send(ctx.args.get("message"))
```

### PR DESCRIPTION
**Pull request by bleudev**
**Description of problem**

Rewite all code in disspy

**For what part of library?**

disspy and documentation

**Checklist of changes**

Closes #93
- [x] abstract.py
- [x] activity.py
- [x] application_commands.py
- [x] channel.py
- [x] client.py
- [x] core.py
- [x] embed.py
- [x] errors.py
- [x] guild.py
- [x] jsongenerators.py
- [x] reaction.py
- [x] thread.py
- [x] typ.py
- [x] ui.py
- [x] user.py
- [x] Tests
- [x] Optimizations
- [x] PyLint fixes

**Additional checklist**
- [x] Fix `NoReturn` -> `None` (closes #91)
- [x] Add SupportsStr to content argument in send(), reply() methods
- [x] Replace DisBotStatus to Literal[...]s
- [x] Create ConnectionState and DisWebhook (old: Flow) object
- [x] Rename core.py to http.py
- [x] Create TypedDicts to application commands, channels, messages and other objects
- [x] Delete all `async with ClientSession()` in HTTP methods except one in `DisBot.__init__()`
- [x] Update code readable
- [x] README.md update
- [x] `__init__.py` update

**Update application commands (Closes #96)**
- [x] @bot.command()
- [x] @bot.context_menu()
- [x] @app_commands.describe()
- [x] @app_commands.options.describe()
- [x] Option object

### CODE REVIEW

**Comment by itttgg** [comment]
G

File: disspy/client.py [hunk 1]
```diff
@@ -600,6 +604,9 @@ def run(self, status: Optional[Union[DisBotStatus, str]] = None,
 
         run(self._runner())
 
+    def _internet_error(self):
+        raise errors.InternetError("Please turn on your internet!", "-1000")
```
Comment (bleudev) [inline disspy/client.py]:
```suggestion
        _m = "Please turn on your internet!"
        _code = "-1000"
        raise errors.InternetError(_m, _code)
```

**Review by bleudev — COMMENTED** [review]
All is normal but there is mini flaws

File: disspy/client.py [hunk 2]
```diff
@@ -600,6 +604,11 @@ def run(self, status: Optional[Union[DisBotStatus, str]] = None,
 
         run(self._runner())
 
+    def _internet_error(self):
```
Comment (bleudev) [inline disspy/client.py]:
```suggestion
    def _raise_internet_error(self):
```

File: disspy/client.py [hunk 3]
```diff
@@ -609,7 +618,7 @@ async def _runner(self) -> NoReturn:
             await self._on_close()
             await self._dissconnenter()
         except requests.exceptions.ConnectionError:
-            raise errors.InternetError("Please turn on your internet!", "-1000")
+            self._internet_error()
```
Comment (bleudev) [inline disspy/client.py]:
```suggestion
            self. _raise_internet_error()
```

**Review by bleudev — COMMENTED** [review]
1 part: reviewed

**Review by bleudev — COMMENTED** [review]
2 part: viewed

File: disspy/client.py [hunk 4]
```diff
@@ -674,20 +704,10 @@ def get_thread(self, thread_id: ThreadId):
...
-        if j["type"] == 11:  # Public thread
-            return DisThread(j, self.token)
+        thread = self.api.get_channel_or_thread(thread_id)
 
-        if j["type"] == 12:  # Private thread
-            return DisPrivateThread(j, self.token)
+        if isinstance(thread, Thread):
+            return thread
 
         raise RuntimeError("This channel is not thread! Use get_channel() method")
```
Comment (bleudev) [inline disspy/client.py]:
```suggestion
       _m = "This channel is not thread! Use get_channel() method"
        raise RuntimeError(_m)
```

File: disspy/client.py [hunk 5]
```diff
@@ -651,17 +684,14 @@ def get_channel(self, channel_id: ChannelId) -> Union[DisChannel, DisDmChannel]:
...
-        _hdrs = {'Authorization': f'Bot {self.token}'}
+        channel = self.api.get_channel_or_thread(channel_id)
 
-        j = get(_u, headers=_hdrs).json()
+        if isinstance(channel, Channel):
+            return channel
 
-        if j["type"] == 1:  # Dm Channels
-            return DisDmChannel(channel_id, self.token)
+        raise RuntimeError("This channel is not channel! Use get_thread() method")
```
Comment (bleudev) [inline disspy/client.py]:
```suggestion
       _m = "This channel is not channel! Use get_thread() method"
        raise RuntimeError(_m)
```

File: disspy/__init__.py [hunk 6]
```diff
@@ -26,10 +26,10 @@
 # Files
 import disspy.activity
 import disspy.abstract
-import disspy.application_commands
+import disspy.app_commands
 import disspy.channel
-import disspy.client
-import disspy.core
+import disspy.bot
+import disspy.http
```
Comment (bleudev) [inline disspy/__init__.py]: Where is import disspy.webhook?

File: disspy/typ.py [hunk 7]
```diff
@@ -54,4 +57,16 @@ def __new__(cls, *args) -> list:
...
+class SupportsStr(Protocol):
+    """SupportsStr
+    Protocol with __str__() method
+    """
+    @abstractmethod
+    def __str__(self) -> str:
+        pass
+
+# New types
+Url     = NewType("Url",         str)
```
Comment (bleudev) [inline disspy/typ.py]:
```suggestion
Url     = str
```

**Review by bleudev — COMMENTED** [review]
Last part: viewed

**Pull request merged**